CorsixTH/Lua/rooms/ward.lua

```diff
@@ -123,37 +124,62 @@ function WardRoom:doStaffUseCycle(humanoid)
   humanoid:queueAction(MeanderAction():setLoopCallback(meanders_loop))
 end
 
+function WardRoom:calculateHealingFactor()
+  local patient_count = 1
+  local nurse_factor = 0
+  for humanoid in pairs(self.humanoids) do
+    if not humanoid:isLeaving() then
+      if class.is(humanoid, Patient) then
+        patient_count = patient_count + 1
+      elseif humanoid.humanoid_class == "Nurse" and not humanoid.fired then
+        nurse_factor = nurse_factor + 0.5 + humanoid:getServiceQuality()
+      end
+    end
+  end
+  -- Difficulty of healing - 10% faster on easier, 10% slower on hard mode
+  local difficulty_factor = 0.8 + self.world.map:getDifficulty() * 0.1
```

Comment:
You can move this inside the `nurse_factor > 0` case, as there is the only place where you need the value.
In that way, you don't waste computing time for values that you eventually don't use.